tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -297,15 +302,16 @@ _load_and_attach_ebpf_programs(_Outptr_ struct bpf_object** return_object)
 
     if (_globals.attach_v6_program) {
         printf("Attaching v6 program\n");
-        bpf_program* connect_program_v6 = bpf_object__find_program_by_name(object, "connect_redirect6");
+        bpf_program* connect_program_v6 = bpf_object__find_program_by_name(object.get(), "connect_redirect6");
         REQUIRE(connect_program_v6 != nullptr);
 
         result = bpf_prog_attach(
             bpf_program__fd(const_cast<const bpf_program*>(connect_program_v6)), 0, BPF_CGROUP_INET6_CONNECT, 0);
         REQUIRE(result == 0);
     }
 
-    *return_object = object;
+    //unique_object.reset(nullptr);
+    *return_object = object.get();
```

Comment:
Should not have commented out code.  I think this should be:
```suggestion
    *return_object = object.get();
    object.reset(nullptr);
```